packages/cli/src/utils/process.ts

```diff
@@ -13,24 +11,22 @@ export type CompilationResult = {
   runners: RunnerInfo[]
 }
 
-export const compile = async (processFilePath: string, ipfsClient: any, lcd: LCD, lcdEndpoint: string, orchestratorEndpoint: string, mnemonic: string, engineAddress: string, env: string[] = []): Promise<CompilationResult> => {
+export const compile = async (processFilePath: string, env: string[], instanceResolver: (definition: { src: string, env: string[] }) => Promise<RunnerInfo>): Promise<CompilationResult> => {
   const runners: RunnerInfo[] = []
-  const instanceReolver = async (instanceObject: any): Promise<string> => {
-    if (!instanceObject.instanceHash && !instanceObject.instance) throw new Error('"instanceHash" or "instance" not found in the process\'s definition')
-    if (instanceObject.instanceHash) return instanceObject.instanceHash
-    const { src, env } = instanceObject.instance
-    const definition = await Service.compile(src, ipfsClient)
-    const service = await Service.create(lcd, definition, mnemonic)
-    const runner = await Runner.create(lcdEndpoint, orchestratorEndpoint, mnemonic, engineAddress, service.hash, env)
-    if (!runners.find(x => x.hash === runner.hash)) runners.push(runner)
-    return runner.instanceHash
-  }
-
 
-  const definition = await compileProcess(readFileSync(processFilePath), instanceReolver, env.reduce((prev, env) => ({
-    ...prev,
-    [env.split('=')[0]]: env.split('=')[1],
-  }), {}))
+  const definition = await compileProcess(
+    readFileSync(processFilePath),
+    async definition => {
+      if (!definition.instanceHash && !definition.instance) throw new Error('"instanceHash" or "instance" not found in the process\'s definition')
+      if (definition.instanceHash) return definition.instanceHash
+      const runner = await instanceResolver(definition.instance)
+      if (!runners.find(x => x.hash === runner.hash)) runners.push(runner)
+      return runner.instanceHash
+    },
+    env.reduce((prev, env) => ({
```

Comment:
```suggestion
    (env || []).reduce((prev, env) => ({
```

Reply:
fixed in the next PR https://github.com/mesg-foundation/js-sdk/pull/238 but can be merged here :)